### PR TITLE
Add dns domain is

### PR DESCRIPTION
**Pull request by fc-rollsd**

### CODE REVIEW

File: src/pac-file-tester.ts [hunk 1]
```diff
@@ -30,11 +30,8 @@ export const testPacFile = async (file: string, url: string, ip: string = addres
 }
 
 export const getHost = (url: string): string => {
-  const regex = /^(?:https?:\/\/)?(?:[^@\n]+@)?(?:www\.)?([^:\/\n]+)/im
-
-  const match = regex.exec(url)
-
-  return match[1]
+  const urlObj = new URL(url);
+  return urlObj.hostname
```
Comment (fc-rollsd) [inline src/pac-file-tester.ts:34]: Not sure why the original strips `www.` but pretty sure we should be checking against the actual hostname, easy to obtain that with built-in URL class.

File: src/pac-file-tester.ts [hunk 2]
```diff
@@ -92,7 +89,12 @@ const vmContext = (ip: string) => {
   }
 
   const isPlainHostName = (host: string) => {
-    return (host.search('\\\\.') == -1);
+    return host.indexOf(".") === -1;
```
Comment (fc-rollsd) [inline src/pac-file-tester.ts:92]: This was not working as expected, added additional test against 
`IS_PLAIN_HOSTNAME_PAC` to reflect working behaviour.

Previously, this always returns TRUE.
```
"localhost".search('\\\\.')
-1
"localhost.com".search('\\\\.')
-1
"www.google.com".search('\\\\.')
-1
"foo".search('\\\\.')
-1
```

File: src/pac-file-tester.ts [hunk 3]
```diff
@@ -92,7 +89,12 @@ const vmContext = (ip: string) => {
...
+  }
+
+  const dnsDomainIs = (host: string, domain: string) => {
+    console.log(`${host} compare to ${domain}`);
```
Comment (fc-rollsd) [inline src/pac-file-tester.ts:96]: Added `dnsDomainIs`  as it was in a proxy file I was testing:

see https://findproxyforurl.com/pac-functions/

File: src/pac-file-tester.ts [hunk 4]
```diff
@@ -92,7 +89,11 @@ const vmContext = (ip: string) => {
   }
 
   const isPlainHostName = (host: string) => {
-    return (host.search('\\\\.') == -1);
+    return host.indexOf(".") === -1;
+  }
+
+  const dnsDomainIs = (host: string, domain: string) => {
+    return host.endsWith(domain);
```
Comment (fc-rollsd) [inline src/pac-file-tester.ts:96]: Added this one as it was in a PAC file I was testing:
https://findproxyforurl.com/pac-functions/

File: src/pac-file-tester.ts (hunk 2 above)
Comment (Arcath) [reply src/pac-file-tester.ts:92]: 🤦‍♂️ Good catch. I'm sure these are the functions I copied straight out of Firefox.

File: src/pac-file-tester.ts (hunk 4 above)
Comment (Arcath) [reply src/pac-file-tester.ts:96]: This is an odd one. checking if `some.domain.tld` ends `domain.tld` is not the same as the way findproxyforurl.com explains it, however it does appear to be the way [browsers implemented it](https://bugs.chromium.org/p/chromium/issues/detail?id=299649#c4).

Little odd but I'd rather have it work this way so it matches browsers.

_this comment is here more for a record incase I get bug reports in the future_

**Comment by Arcath** [comment]
Thanks for the PR. Not sure how `dnsDomainIs` slipped through the net.

I'll merge and release this.

**Pull request merged**